app/src/main/java/com/asfoundation/wallet/ui/iab/PaymentMethodsPresenter.kt

```diff
@@ -1000,6 +1002,13 @@ class PaymentMethodsPresenter(
     )
   }
 
+  private fun handleCurrencyChanges(price: Value) {
```

Comment:
When the currency changes, you need to use the new currency to make the payment. It's using the cachedFiatValue to make the purchase, but it should use the currency from the specific payment method